gradle/project.gradle.kts

```diff
@@ -89,6 +95,8 @@ dependencies {
     // Kotlin
     compile(kotlin("stdlib-jdk8"))
     compile(kotlin("reflect"))
+    compile("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.0.0")
+    compile("org.jetbrains.kotlinx:kotlinx-coroutines-javafx:1.0.0")
```

Comment:
Agree wit the 3 style.
Disagree on transitiveness. We need to be explicit, and with dependencies that applies two fold. I don't care what relationship is there between those two, my application depends on both. If in the future it will only depend on the core, removing javafx would also remove the core, which is absolutely not what we want. I think this is explicit, clear and correct.

Reply:
In other words im not bringing core jsut to satisfy javafx, Im bringing both of them, the fact that javafx uses core is of no importance to me and remains a coincidence. It can be looked at this way: core is depended upon twice, by javafx and the application. If it was only depended on by the javaf, then is hould not be in our build file.

So the rule is to express only direct dependencies, what is exactly the case here. Its just there is also inter-dependencies between our dependencies.

Reply:
But this is a special case where one library is directly built on top of another. I totally get your point, I guess it's just personal preference.

Reply:
Yes variable is definitely going to happen.

I argue however that the decision about the dependency is a requirement not a subjective opinion. A project must express its direct dependencies. Its dependencies' interrelations are of no consequence.

Reply:
Also, Gradle 5 will introduce _dependency version alignment_: https://docs.gradle.org/5.0-milestone-1/release-notes.html. I am not yet completely sure what it does, but it may be helpful.

But I just found out about a much easier way, PR coming shortly...